relays/bin-ethereum/src/ethereum_exchange.rs

```diff
@@ -67,7 +67,7 @@ pub struct EthereumExchangeParams {
 	/// Relay working mode.
 	pub mode: ExchangeRelayMode,
 	/// Metrics parameters.
-	pub metrics_params: Option<MetricsParams>,
+	pub metrics_params: MetricsParams,
```

Comment:
`Option<S>` != `S { Option<>, Option<> }`. Fields may be `None` because of different reasons:
1) `address` is `None` when you don't need to expose metrics - either user has disabled them, or they are exposed somewhere else;
2) `registry` is `Some` when it's either shared by multiple loops, or some metrics were pre-registered. If it's `None` you're the only loop and shall use your own prefix when registry is created.

The structure has `disabled()` method, which may be used e.g. by tests. In binaries you first deal with `MetricsAddress`, which is both `Default` and is passed inside `Option<>` + there's `impl From<Option<MetricsAddress>> for MetricsParams`

Reply:
Fair enough.